### ISSUE TITLE
cliConfig resource and runCI job

### ISSUE THREAD

**Issue by seniorquico** [Shippable/support#3983]
I have a `cliConfig` resource with an `azure-keys` integration.

I connected the `cliConfig` resource to a `runSh` job in our assembly line configuration. An Azure CLI login step is automatically executed before our custom script steps are executed. 👍

I connected the `cliConfig` resource to a `runCI` job in our assembly line configuration. However, an Azure CLI login step is **not** automatically executed. I had to manually script the Azure CLI login using the environment variables provided by the integration. 😕

Should the `cliConfig` resource automatically login to Azure in our `runCI` job? Is this a limitation of the new `azure-keys` integration? Of note, the `azure-keys` integration isn't listed on the [`cliConfig` resource docs](http://docs.shippable.com/platform/workflow/resource/cliconfig/).

**Comment by trriplejay** [comment]
It appears that the docs are incorrect on this.  cliConfigs are not supported in runCI type jobs.  (you can add them as an IN, but as you saw, they do not perform their primary function).

I think what you did as a workaround is the best way to use azure cli in runCI at this time.

I'll open an issue to update the docs, but also to natively support azure in runCI.  For many of our integrations, you can add them directly to your shippable.yml in the 'integrations' section.  In some cases this will automatically configure the tooling associated with the integration, but this is not the case yet for azure keys.  I'll mark this as a feature request.

**Comment by BryanCrotaz** [comment]
for clarity for users, resources should work the same everywhere. All cliConfigs should work automatically wherever they can be used.

**Comment by BryanCrotaz** [comment]
Having just banged my head on the desk for a couple of hours, and then come across this issue from before Xmas, I can confirm that AWS keys don't work either.

**Comment by BryanCrotaz** [comment]
Note: http://docs.shippable.com/platform/workflow/resource/cliconfig/ has not been updated as described above in November

**Comment by seniorquico** [comment]
@BryanCrotaz It has... sorta(?)... been fixed in their documentation. I'll admit, it's still confusing.

https://github.com/Shippable/docs/pull/790

@trriplejay The "Used in Jobs" section indicates a `cliConfig` resource may only be used as an `IN` to a `runSh` job. Read as-is, that isn't technically correct. Technically, it may be used as an `IN` to both `runSh` and `runCI` jobs. In fact, the very next section, "Default Environment Variables", and your prior comment confirms this. I've had to interpret... and that's what makes this still confusing... the "Used in Jobs" section as meaning "the automated login behavior is only applied in `runSh` jobs". I would strongly advocate that this just get spelled out in plain English (and this would apply to other, similar pages in the Shippabel docs, too).

**Comment by BryanCrotaz** [comment]
I would say save time spelling it out and spend the time fixing it :)

@ttrahan

**Comment by seniorquico** [comment]
My point was that this potential confusion isn't isolated to this one resource type... this same language is used on other pages, too.

**Comment by manishas** [comment]
We'll clean up those docs and also look at other resource types to make sure it's super clear where each resource is supported.
